steppy/base.py

```diff
@@ -367,18 +376,24 @@ def transform(self, data):
                 step_inputs = self._adapt(step_inputs)
             else:
                 step_inputs = self._unpack(step_inputs)
-            step_output_data = self._cached_transform(step_inputs)
+            step_output_data = self._transform_operation(step_inputs)
         return step_output_data
 
-    def clean_cache(self):
-        """Removes everything from the directory ``<experiment_directory>/cache``.
+    def clean_cache_step(self):
+        """Clean cache for current step.
+        """
+        logger.info('Step {}, cleaning cache'.format(self.name))
+        self.output = None
+        gc.collect()
+
+    def clean_cache_pipeline(self):
+        """Clean cache for all steps that are upstream to `self`.
         """
-        logger.info('cleaning cache...')
-        paths = glob.glob(os.path.join(self.exp_dir_cache, '*'))
-        for path in paths:
-            logger.info('removing {}'.format(path))
-            os.remove(path)
-        logger.info('cleaning cache done')
+        logger.info('Cleaning cache for the entire upstream pipeline')
+        for step in self.all_steps.values():
+            logger.info('Step {}, cleaning cache'.format(step.name))
+            step.output = None
+            gc.collect()
```

Comment:
@kamil-kaczmarek I am not 100% that gc.collect() will change anything here. I know that it works with del obj; gc.collect() but when simply changing value to None idk. We should probably test it.

Reply:
corrected.